File: provaPointC/provaPointC/provaPointC/QuadraticEquation.vb
Public Module QuadraticEquation
    Private Function delta(_a As Double, _b As Double, _c As Double)
        Return Math.Pow(_b, 2) - 4 * _a * _c
    End Function

    Public Function solve(_a As Double, _b As Double, _c As Double, _sign As Integer)
        Return (-_b + Math.Sign(_sign) * Math.Sqrt(Math.Abs(delta(_a, _b, _c)))) / (2 * _a)
    End Function
End Module

File: CCS_workspace/ProvaParser/source/QuadraticEquation.c
#include "QuadraticEquation.h"

float delta(float _a, float _b, float _c)
{
  return pow(_b, 2) - 4 * _a * _c;
}

float solveQE(float _a, float _b, float _c, bool _isPositive)
{
  if (_isPositive)
  {
    return (-_b + sqrt(delta(_a, _b, _c))) / (2 * _a);
  }
  else
  {
    return (-_b - sqrt(delta(_a, _b, _c))) / (2 * _a);
  }
}
